### PR TITLE
Fix AIX build of PAL because of GCC system header mangling

### PR DESCRIPTION
**Pull request by NattyNarwhal**
GCC's "fix"includes tool claims another function as its victim. Import the func definition from the headers; we can't trust GCC to not replace them with "fixed" versions.

(FreeBSD people: `_WITH_GETLINE` to replace what was removed would help you)

### CODE REVIEW

**Comment by NattyNarwhal** [comment]
Wait, this may be slightly erroneous. Let me refer to the headers again; something doesn't seem right...

**Comment by NattyNarwhal** [comment]
It's mergeable; per the comment in why this BS exists:

```c
// GCC "fix"includes removes this definition from stdio.h entirely, even if you 
// opt into using getline via the `_GETDELIM` definition. (as it can conflict 
// with some existing code, so it's hidden behind a define) Adding to the 
// tragedy, the define we need for the runtime (`_ALL_SOURCE`) sets the other 
// necessary definition (`_XOPEN_SOURCE=700`) to 600, overriding it. As such, 
// redefine getline function declaration here and cut the gordian knot caused 
// by commercial Unix and GNU. 
```

File: src/Native/Unix/System.Native/pal_io.c [hunk 1]
```diff
@@ -47,6 +42,14 @@
 // Somehow, AIX mangles the definition for this behind a C++ def
 // Redeclare it here
 extern int     getpeereid(int, uid_t *__restrict__, gid_t *__restrict__);
+// GCC "fix"includes removes this definition from stdio.h entirely, even if you
+// opt into using getline via the `_GETDELIM` definition. (as it can conflict
+// with some existing code, so it's hidden behind a define) Adding to the
+// tragedy, the define we need for the runtime (`_ALL_SOURCE`) sets the other
+// necessary definition (`_XOPEN_SOURCE=700`) to 600, overriding it. As such,
+// redefine getline function declaration here and cut the gordian knot caused
+// by commercial Unix and GNU.
```
Comment (stephentoub) [inline src/Native/Unix/System.Native/pal_io.c]: This is a fairly emotional comment :)  Can we trim it back to just the facts?

**Comment by stephentoub** [comment]
> be kinder and to the point

Thanks ;)

**Pull request merged**